lib/matplotlib/collections.py

```diff
@@ -392,8 +392,8 @@ def draw(self, renderer):
             else:
                 combined_transform = transform
             extents = paths[0].get_extents(combined_transform)
-            if (extents.width < self.figure.bbox.width
-                    and extents.height < self.figure.bbox.height):
+            if (extents.width < self.get_figure(root=True).bbox.width
+                    and extents.height < self.get_figure(root=True).bbox.height):
```

Comment:
I'm also not sure why we are looking at the total size not what it might be clipped to, but it has been like this from 2014 so :+1: on leaving it unless we understand why it should be changed to the subfigure.